File: domain/src/main/kotlin/com/readler/domain/usecase/GetLibraryBooksUseCase.kt
package com.readler.domain.usecase

import com.readler.core.model.BookMetadata
import com.readler.domain.repository.BookRepository
import kotlinx.coroutines.flow.Flow

class GetLibraryBooksUseCase(
    private val bookRepository: BookRepository
) {
    operator fun invoke(): Flow<List<BookMetadata>> = bookRepository.observeLibrary()
}

File: lib/domain/usecase/get_library_books_use_case.dart
import '../../core/model/book_metadata.dart';
import '../repository/book_repository.dart';

class GetLibraryBooksUseCase {
  final BookRepository _bookRepository;

  GetLibraryBooksUseCase(this._bookRepository);

  Stream<List<BookMetadata>> call() => _bookRepository.observeLibrary();
}
